### PR TITLE
Fixed display name to be course_name from full_name

### LINKED ISSUE #257: Bug — Courses not in catalog not available in Course search
**Describe the bug**
Any course that is not in the RPI catalog (mainly topics courses) do not have a full_name column in the parser, so they are not being displayed in the course search on the front end.

**To Reproduce**
1. Load the yacs environment
2. Load the Spring 2021 CSV
3. Add the course COGS-6960
4. Try to add the course COGS-6961
5. Notice that the name of COGS-6960 is correct on the schedule and incorrect in course search
6. Notice that COGS-6961 cannot be added from course search, but is in the csv

**Expected behavior**
COGS-6960, 6961, 6962, & 6963 should be able to be added in course search. Their names in both course search and on the schedule should be the same as the course_name column in the spring 2021 csv,

**Screenshots**
Screenshot 2020-10-27 205303
Screenshot 2020-10-27 205321

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - Version: ??

**Additional context**
The full_name column is needed for display in the Explorer, but should not be relevant to course search at all.

### PR DESCRIPTION
**Pull request by marchdan**
**Issue**

Closes #257 

**Test Procedure**

*Example:*
> 1. Load the environment
> 2. Load spring-2021 in the admin panel
> 3. Try to schedule a topics course such as COGS 4960 or COGS 6960
> 4. See that the name on course search matches the name on the calendar matches the course name on SIS even though it's in the catalog under a different name.

**Photos**
Before:
names-before

Now:
names-after

**Additional Info**

N/A

### CODE REVIEW

File: src/web/src/components/CourseListing.vue [hunk 1]
```diff
@@ -10,7 +10,7 @@
         {{ readableDate(course.date_end) }})
         <br />
         {{
-          (course.full_title && course.full_title.toUpperCase()) || course.title
+          course.title || (course.full_title && course.full_title.toUpperCase())
```
Comment (123joshuawu) [inline src/web/src/components/CourseListing.vue:13]: I'm a little confused on what this is fixing, if `course.full_title` does not exist, shouldn't it default to `course.title`? Or is `course.full_title` incorrect?

Comment (jshom) [reply src/web/src/components/CourseListing.vue:13]: Ye same question, is there a specific course where title is more descriptive than full_title?

Comment (marchdan) [reply src/web/src/components/CourseListing.vue:13]: The exact example given in the screenshots. The "Topics in Cognitive Science" course will always have that name in the catalog as a place holder. But the actual course name is Social Computing.

Comment (marchdan) [reply src/web/src/components/CourseListing.vue:13]: In theory, course.full_title shouldn't be used here at all, but I wanted to leave the logic that was already in place

Comment (marchdan) [reply src/web/src/components/CourseListing.vue:13]: course.title is course_name in the csv, i.e. the real name of the course from SIS

**Comment by marchdan** [comment]
Also just noticed another issue, 4961, 4962, 4963 aren't showing up in the course listings on the screenshots...

File: src/web/src/components/CourseListing.vue (hunk 1 above)
Comment (123joshuawu) [reply src/web/src/components/CourseListing.vue:13]: Hm, so it seems like we should just use `course.title`

Comment (marchdan) [reply src/web/src/components/CourseListing.vue:13]: That was my thought

Comment (marchdan) [reply src/web/src/components/CourseListing.vue:13]: I'll double check what the schedule blocks use

**Review by 123joshuawu — APPROVED** [review]

**Comment by jshom** [comment]
```
    "corequisites": null,
    "date_end": "Fri, 18 Dec 2020 00:00:00 GMT",
    "date_start": "Mon, 31 Aug 2020 00:00:00 GMT",
    "department": "ARCH",
    "description": "Building on the knowledge, critical perspectives, and abilities that students have acquired in the previous courses in the history, theory, and criticism sequence, this course begins a study of the ideas, values, theories, and practices that contributed to the rise of modernity in the western world and eventually on a global scale. This course explores the ideal of modernity as both a cultural phenomenon and as a technological enframing of the world, scientific rationality, historical consciousness, etc.",
    "frequency": "Fall term annually.",
    "full_title": "Cloud Atlas: 20th Century Architecture, Culture and Civilization",
    "level": 4120,
    "name": "ARCH-4120",
    "prerequisites": ["ARCH-4100"],
    "raw_precoreqs": "ARCH 4100 An Architectural Genealogy.",
    "school": "Architecture",
    "sections": [{
        "crn": "30481",
        "department": "ARCH",
        "level": 4120,
        "semester": "FALL 2020",
        "sessions": [{
            "crn": "30481",
            "day_of_week": 4,
            "location": null,
            "section": "02",
            "semester": "FALL 2020",
            "time_end": "13:20:00",
            "time_start": "12:00:00"
        }, {
            "crn": "30481",
            "day_of_week": 1,
            "location": null,
            "section": "02",
            "semester": "FALL 2020",
            "time_end": "13:20:00",
            "time_start": "12:00:00"
        }]
    }, {
        "crn": "29197",
        "department": "ARCH",
        "level": 4120,
        "semester": "FALL 2020",
        "sessions": [{
            "crn": "29197",
            "day_of_week": 4,
            "location": null,
            "section": "01",
            "semester": "FALL 2020",
            "time_end": "13:20:00",
            "time_start": "12:00:00"
        }, {
            "crn": "29197",
            "day_of_week": 1,
            "location": null,
            "section": "01",
            "semester": "FALL 2020",
            "time_end": "13:20:00",
            "time_start": "12:00:00"
        }]
    }, {
        "crn": "30482",
        "department": "ARCH",
        "level": 4120,
        "semester": "FALL 2020",
        "sessions": [{
            "crn": "30482",
            "day_of_week": 1,
            "location": null,
            "section": "03",
            "semester": "FALL 2020",
            "time_end": "13:20:00",
            "time_start": "12:00:00"
        }, {
            "crn": "30482",
            "day_of_week": 4,
            "location": null,
            "section": "03",
            "semester": "FALL 2020",
            "time_end": "13:20:00",
            "time_start": "12:00:00"
        }]
    }],
    "semester": "FALL 2020",
    "title": "CLOUD ATLAS: 20TH CENTURY ARCH"
```
Take for instance this class, this is why we are using `full_title`, I think some other courses also have their names trimmed a bit in `title`.
I'm fine with changing it, just making sure we're on the same page with why it was there in the beginning.

**Comment by marchdan** [comment]
> ```
>     "corequisites": null,
>     "date_end": "Fri, 18 Dec 2020 00:00:00 GMT",
>     "date_start": "Mon, 31 Aug 2020 00:00:00 GMT",
>     "department": "ARCH",
>     "description": "Building on the knowledge, critical perspectives, and abilities that students have acquired in the previous courses in the history, theory, and criticism sequence, this course begins a study of the ideas, values, theories, and practices that contributed to the rise of modernity in the western world and eventually on a global scale. This course explores the ideal of modernity as both a cultural phenomenon and as a technological enframing of the world, scientific rationality, historical consciousness, etc.",
>     "frequency": "Fall term annually.",
>     "full_title": "Cloud Atlas: 20th Century Architecture, Culture and Civilization",
>     "level": 4120,
>     "name": "ARCH-4120",
>     "prerequisites": ["ARCH-4100"],
>     "raw_precoreqs": "ARCH 4100 An Architectural Genealogy.",
>     "school": "Architecture",
>     "sections": [...],
>     "semester": "FALL 2020",
>     "title": "CLOUD ATLAS: 20TH CENTURY ARCH"
> ```
> 
> Take for instance this course, this is why we are using `full_title`, I think some other courses also have their names trimmed a bit in `title`.
> I'm fine with changing it, just making sure we're on the same page with why it was there.
> 
> More I think of it, it seems like an RPI issue, maybe when we parse the registrar xml for the job, any course with Topics in.. is set to null since I think its the default RPI name for 4XXX/6XXX courses.
> 
> What do you think? @123joshuawu @marchdan

So my way of thinking is that RPI names those courses that way on SIS to limit character count, which I think might be good for us also. Plus I would much rather a course show with a shortened name like the example than the wrong name like I mentioned earlier.

I don't have an issue with setting it to null, but I don't really want to rely on the accuracy of the registrar for this to work. I'd prefer just using the course_title, especially since that is what we display on the calendar (which is why the issue came up in the first place)

@jshom

**Review by jshom — APPROVED** [review]

**Pull request merged**